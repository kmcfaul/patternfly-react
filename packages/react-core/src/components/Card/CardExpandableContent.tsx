import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/Card/card';
import { css } from '@patternfly/react-styles';

export interface CardExpandableContentProps extends React.HTMLProps<HTMLDivElement> {
  /** Content rendered inside the Card Body */
  children?: React.ReactNode;
  /** Additional classes added to the Card Body */
  className?: string;
}

export const CardExpandableContent: React.FunctionComponent<CardExpandableContentProps> = ({
  children = null,
  className = '',
  ...props
}: CardExpandableContentProps) => (
  <div className={css(styles.cardExpandableContent, className)} {...props}>
    {children}
  </div>
);
CardExpandableContent.displayName = 'CardExpandableContent';
