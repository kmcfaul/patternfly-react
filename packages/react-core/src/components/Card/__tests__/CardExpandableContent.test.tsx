import React from 'react';
import { CardExpandableContent } from '../CardExpandableContent';
import { shallow } from 'enzyme';

test('renders successfully', () => {
  const view = shallow(<CardExpandableContent />);
  expect(view).toMatchSnapshot();
});
