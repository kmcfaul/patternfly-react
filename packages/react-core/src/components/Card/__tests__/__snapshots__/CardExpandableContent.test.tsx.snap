// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<div
  className="pf-c-card__expandable-content"
/>
`;
